crates/ef-testing-proc-macro/src/content_reader.rs

```diff
@@ -0,0 +1,54 @@
+use std::collections::BTreeMap;
+
+use serde_json::Value;
+
+use crate::{dir_reader::PathWrapper, utils::blockchain_tests_to_general_state_tests_path};
+
+pub struct ContentReader;
+
+impl ContentReader {
+    pub fn secret_key(path: PathWrapper) -> Result<Value, eyre::Error> {
+        let path = blockchain_tests_to_general_state_tests_path(path);
+        let content = path.read_file_to_string()?;
+
+        let cases: BTreeMap<String, Value> = serde_json::from_str(&content)?;
+        let case = cases
+            .into_values()
+            .collect::<Vec<_>>()
+            .first()
+            .ok_or_else(|| eyre::eyre!("No case found"))?
+            .clone();
+
+        Ok(case
+            .get("transaction")
```

Comment:
can it happen that there is more than one transaction key?

Reply:
I don't think you have a valid json if you have more than once the same key right?

Reply:
I guess then the question becomes, is there a situation where there are multiple cases in any given file in GeneralStateTests?

Reply:
There might be, but since we are only taking the transaction to extract the secret key, there hasn't been an issue up to now.